### PR TITLE
style: update inline code border and bg color

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksupdate-inline-code-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1203652232454021/1204108582255289) 🎟️
- [Figma File](https://www.figma.com/file/v0Hyqvjdf6QJAvRkdIBV24/Dark-Mode?node-id=302-30801&t=zKEXDsfXxqaz5rY9-0) 🎨 

## 🗒️ What

Updates the inline code background color to be a semantic token instead of the base neutral token. Also adjusts the border to use a token. 

Screenshot 2023-03-16 at 10 25 11 AM

### Regular inline code
- [Visit a page](https://dev-portal-git-ksupdate-inline-code-hashicorp.vercel.app/terraform/cli/commands/init) with inline code
- [And another](https://dev-portal-git-ksupdate-inline-code-hashicorp.vercel.app/terraform/enterprise/run/cli#excluding-files-from-upload)
- Validate that light mode looks the same as production
- Check dark mode, ensure the background and border colors matches [the spec](https://www.figma.com/file/v0Hyqvjdf6QJAvRkdIBV24/Dark-Mode?node-id=302-30801&t=zKEXDsfXxqaz5rY9-0) 

### Inline code in legacy alerts
- [Visit a page](https://dev-portal-git-ksupdate-inline-code-hashicorp.vercel.app/terraform/enterprise/run/run-environment#state-access-and-authentication) with the legacy alert
- Validate that the background and border colors for the inline code match [the spec](https://www.figma.com/file/v0Hyqvjdf6QJAvRkdIBV24/Dark-Mode?node-id=302-30801&t=zKEXDsfXxqaz5rY9-0) in both dark and light mode. 

## 💭 Anything else?

Waiting on updated blue values to finish this override.

### CODE REVIEW

**Comment by kendallstrautman** [comment]
@braydenlove pinging you to check out the updated border color!

**Comment by kendallstrautman** [comment]
@braydenlove also the background color token in the ticket is darker than whats in prod for light mode, do you want that just for dark mode or to apply to both?

File: src/components/dev-dot-content/mdx-components/mdx-inline-code/mdx-inline-code.module.css [hunk 1]
```diff
@@ -4,11 +4,18 @@
...
-	border: 1px solid rgba(101, 106, 118, 0.2);
+	border: 1px solid var(--token-color-palette-alpha-200);
 	line-height: 1;
 	padding: 2px 4px 3px 4px;
+
+	@nest html[data-theme='dark'] & {
+		background-color: var(--token-color-surface-strong);
+
+		/* --token-color-border-strong */
+		border-color: var(--token-color-border-primary);
```
Comment (kendallstrautman) [inline src/components/dev-dot-content/mdx-components/mdx-inline-code/mdx-inline-code.module.css:17]: I'm pretty sure this border token accounts for the alpha value.

File: src/components/dev-dot-content/mdx-components/mdx-inline-code/mdx-inline-code.module.css [hunk 2]
```diff
@@ -4,11 +4,18 @@
...
 	border-radius: 5px;
-	border: 1px solid rgba(101, 106, 118, 0.2);
+	border: 1px solid var(--token-color-palette-alpha-200);
 	line-height: 1;
 	padding: 2px 4px 3px 4px;
+
+	@nest html[data-theme='dark'] & {
+		background-color: var(--token-color-surface-strong);
+
+		/* --token-color-border-strong */
```
Comment (kendallstrautman) [inline src/components/dev-dot-content/mdx-components/mdx-inline-code/mdx-inline-code.module.css:16]:
```suggestion
```

**Comment by braydenlove** [comment]
@kendallstrautman you nailed the overrides for dark mode! Thanks!

Final thing we need to sort out is the themed border css on the old alerts. I think we should just remove this for both dark and light if possible as it looks better in both scenarios

Context (themed border weirdness):
image

Dark mode today:
image

Dark mode refinement (remove themed css):
image

Light mode today:
image


Light mode refinement (removed themed css):
image

I think having the border helps the inline code stand out in both scenarios! Lmk what you think otherwise this is ✅

File: src/components/dev-dot-content/dev-dot-content.module.css [hunk 3]
```diff
@@ -34,9 +34,13 @@
...
 			& :not(pre) > code {
-				background-color: var(--token-color-palette-neutral-0);
+				background-color: var(--token-color-surface-faint);
 				border-radius: 5px;
-				border: var(--themed-border);
+				border: 1px solid var(--token-color-border-primary);
+
+				@nest html[data-theme='dark'] & {
+					background-color: var(--token-color-surface-strong);
+				}
```
Comment (kendallstrautman) [inline src/components/dev-dot-content/dev-dot-content.module.css:43]: This is the inline code styles for the legacy alert components.

**Review by zchsh — APPROVED** [review]

**Pull request merged**